### PR TITLE
Issue #116 Include layer information to stanford.asdb AS categories

### PR DESCRIPTION
**Pull request by JustinLoye**
## Description

Added a `layer`  property to links  -[r:CATEGORIZED {reference_name: "stanford.asdb"}]-
Added a `PART_OF` relationship  (subcategory) -> (category)
Removed `Other` and `other` subcategories

## Motivation and Context

Issue 116

## How Has This Been Tested?

Thorough check of my file parsing (noticed new errors).

Overall number of links  -[r:CATEGORIZED {reference_name: "stanford.asdb"}]-:
before 394 834, now roughly twice more  592 754

Pushed and observed changes on my local iyp

## Screenshots (if appropriate):

Screenshot from 2024-02-16 14-08-46

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: iyp/crawlers/stanford/asdb.py [hunk 1]
```diff
@@ -53,25 +54,61 @@ def run(self):
...
         # Compute links
         links = []
-        for (asn, category) in lines:
+        for (asn, layer, category) in lines:
 
             asn_qid = asn_id[asn]
             category_qid = category_id[category]
 
-            links.append({'src_id': asn_qid, 'dst_id': category_qid, 'props': [self.reference]})  # Set AS category
+            links.append({'src_id': asn_qid, 'dst_id': category_qid, 'props': [
```
Comment (m-appel) [inline iyp/crawlers/stanford/asdb.py:110]: Can you break this line before `'props'`? :)

**Review by m-appel — CHANGES_REQUESTED** [review]
Pretty much good to go, functionally no problems, only two nitpicks (I would not have mentioned the linebreak, but since you'll touch the file anyways for the other comment I mentioned it).

Btw. I prefer using something like
```python
if not category:
    continue
asns.add(asn)
```
this prevents having so many nested levels of code. But purely stylistic, you don't have to change this.

**Comment by JustinLoye** [comment]
As a side comment, I checked that the PART_OF links are well behaved 

MATCH (n: Tag)-[r:PART_OF {reference_name: "stanford.asdb"}]-(m: Tag) return n, r, m
Should return ${number_of_layer1_categories} graphs of diameter 2.

As shown in this screenshot, it's almost the case
Screenshot from 2024-02-16 17-12-29

the exception (left-most graph) being caused by the error mentioned in https://github.com/InternetHealthReport/internet-yellow-pages/issues/116#issuecomment-1920688892

**Review by m-appel — APPROVED** [review]
Thanks looks good to me.

**Pull request merged**